src/dds/rs-dds-sensor-proxy.cpp

```diff
@@ -401,23 +401,27 @@ void dds_sensor_proxy::add_frame_metadata( frame * const f, nlohmann::json && dd
     f->additional_data.timestamp;
     rsutils::json::get_ex( md_header, timestamp_domain_key, &f->additional_data.timestamp_domain );
 
-    // Expected metadata for all depth images
-    rsutils::json::get_ex( md_header, depth_units_key, &f->additional_data.depth_units );
-
-    // Other metadata fields. Metadata fields that are present but unknown by librealsense will be ignored.
-    auto & metadata = reinterpret_cast< metadata_array & >( f->additional_data.metadata_blob );
-    for( size_t i = 0; i < static_cast< size_t >( RS2_FRAME_METADATA_COUNT ); ++i )
+    if( ! md.empty() )
     {
-        auto key = static_cast< rs2_frame_metadata_value >( i );
-        std::string const & keystr = librealsense::get_string( key );
-        try
+        // Other metadata fields. Metadata fields that are present but unknown by librealsense will be ignored.
+        auto & metadata = reinterpret_cast< metadata_array & >( f->additional_data.metadata_blob );
+        for( size_t i = 0; i < static_cast< size_t >( RS2_FRAME_METADATA_COUNT ); ++i )
         {
-            metadata[key] = { true, rsutils::json::get< rs2_metadata_type >( md, keystr ) };
-        }
-        catch( nlohmann::json::exception const & )
-        {
-            // The metadata key doesn't exist or the value isn't the right type... we ignore it!
-            // (all metadata is not there when we create the frame, so no need to erase)
+            auto key = static_cast< rs2_frame_metadata_value >( i );
+            std::string const & keystr = librealsense::get_string( key );
+            try
+            {
+                if( auto value = rsutils::json::nested( md, keystr ) )
+                {
+                    if( value->is_number_integer() )
```

Comment:
If we will get something else, don't we want to infrom the user?

Reply:
No. This is metadata; we ignore anything we cannot parse.
Same behavior as before -- just doesn't throw exceptions as much (which in Debug mode are constantly triggered).

Reply:
No, the exceptions were when we just didn't *find* the values for the metadata